### ISSUE TITLE
[mkauer on 2017-01-10 22:03:26] : GPS failure gave strange stop times for SPE/MPE moni20 quantities

### ISSUE THREAD

**Issue by dglo** [dglo/pdaq#2777]
When we lost power to GPS and run 128987 crashed, the SPE and MPE moni20 quantities had endpoints way in the future. I'm not sure if this will be reproducible or not but figured it was worth looking into.

**Comment by dglo** [comment]
[tbendfelt on 2017-01-11 22:31:52]
I took a look into this.  The bad timestamp is likely a result of corrupted messages coming from the DOM. If I had to guess I would say the corruption is happening at the DOR card because it is most likely to be affected from a master clock power cycle when the 10MHz input is unstable. In any event,  pDAQ is falling down because it is seeing all sorts of bad behavior from the DOM communications channels.

Despite the fact that the master clock was involved, I am blaming corrupted messages rather than inaccurate time reconstruction.  There are many clues in the logs to support this conclusion.

1) StringHub starts dropping DOMS due to non-sensical messages:
 stringHub#1 icecube.daq.domapp.dataprocessor.AsynchronousDataProcessor-Processor-20A ERROR [2016-12-31 01:03:30.792] Processing encountered an error, will force a shutdown.
java.lang.IllegalArgumentException
	at java.nio.Buffer.limit(Buffer.java:275)
	at icecube.daq.domapp.dataprocessor.HitProcessor.process(HitProcessor.java:119)
	at icecube.daq.domapp.dataprocessor.SynchronousDataProcessor.process(SynchronousDataProcessor.java:85)
	at icecube.daq.domapp.dataprocessor.AsynchronousDataProcessor$4.call(AsynchronousDataProcessor.java:337)
	at icecube.daq.domapp.dataprocessor.AsynchronousDataProcessor$4.call(AsynchronousDataProcessor.java:331)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

2) StringHub also sees out-of-order data:
stringHub#1 icecube.daq.domapp.dataprocessor.UTCDispatcher-Processor-31B ERROR [2016-12-31 01:04:33.133] Out-of-order HIT: last-utc [315398720834464647] current-utc [315398471631654256] last-dom-clock [82550285792716] current-dom-clock [82549288955851] utc-diff [249202810391] (occurrence: 1, reason: Non-Contiguous HIT stream from DOM, action: drop)


3) The mini records feeding this counter also start becoming corrupted:


secondaryBuilders icecube.daq.secBuilder.MoniAnalysis-HKN1Splicer+icecube.daq.secBuilder.MoniAnalysis@60856c9a ERROR [2016-12-31 01:03:52.805] Ignoring fast monitoring record (#flds != 5): F 242 T????
secondaryBuilders icecube.daq.secBuilder.MoniAnalysis-HKN1Splicer+icecube.daq.secBuilder.MoniAnalysis@60856c9a ERROR [2016-12-31 01:04:54.440] Ignoring fast monitoring record (#flds != 5): F 371 T???
secondaryBuilders icecube.daq.secBuilder.MoniAnalysis-HKN1Splicer+icecube.daq.secBuilder.MoniAnalysis@60856c9a ERROR [2016-12-31 01:04:55.479] Ignoring fast monitoring record (#flds != 5): F 471 T:??
